tee_supplicant/src/ree_fs.c

```diff
@@ -108,30 +155,46 @@ static int tee_fs_open(size_t num_params, struct tee_param *params,
 	}
 	strncat(path, name, PATH_MAX);
 
-	fd = new_file(&file);
+	fd = internal_open(path, flags);
 	if (fd < 0) {
-		rc = TEEC_ERROR_GENERIC;
-		goto free;
-	}
-
-	rc = fs_open(file, path, flags);
-	if (rc < 0) {
-		if (rc == -ENOENT) {
-			rc = TEEC_ERROR_ITEM_NOT_FOUND;
-			goto free;
+		if (flags & FS_O_CREATE) {
+			char *dir;
+
+			dir = dname(path);
+			if (!dir) {
+				rc = TEEC_ERROR_GENERIC;
+				goto out;
+			}
+			rc = fs_mkdir(dir);
```

Comment:
It this function recursive? E.g. will it create both "tee" and "1" for path "/tee/1/abc"?

Reply:
This patch removed